### PR TITLE
feat: display details title if defined

### LINKED ISSUE #836: Add a title to the technical details in a resultlist
image

- [ ] a new `fieldsConfiguration.detailsTitleTemplate` input should be added to the component where we set the value of this title
- [ ] (input name to be validated with the team if this one doesn't seem suitable)
- [ ] This template may contain one or several fields to be replace on the fly
- [ ] see `fieldsConfiguration.urlThumbnailTemplate`
- [ ] For reminder, the `fieldsConfiguration` should be iso to `fieldsConfiguration` in the contributor

### PR DESCRIPTION
**Pull request by sebbousquet**
- Fix #836

Grid mode :
Capture d’écran 2024-07-22 à 14 45 28
Capture d’écran 2024-07-22 à 14 45 25

List mode:
Capture d’écran 2024-07-22 à 14 45 42

### CODE REVIEW

File: projects/arlas-components/src/lib/components/results/model/item.ts [hunk 1]
```diff
@@ -114,6 +114,14 @@ export class Item {
...
    */
   public color: string;
+  /**
+    * @description Title for the details section.
+    */
+  public detailsTitle: string;
+  /**
+   * @description If details item title is enabled
+   */
+  public detailTitleEnabled: boolean;
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/results/model/item.ts:124]: In one property you wrote detail**s** and the other just detail. Could it be uniformized ?

File: projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
-  <span *ngFor="let group of getGroups()">
-    <span class="group-item-title {{group.group.toLowerCase().replace(' ','-')}}-item-detail">{{group.group | translate}}</span>
-    <br>
-    <span *ngFor="let detail of group.details" class="detail-item {{detail.key}}-item-detail">
-      <span class="resultdetaileditem-key">{{detail.key | translate}}</span> :
-      <span class="resultdetaileditem-value">{{detail.value | formatNumber:NUMBER_FORMAT_CHAR}}</span>
+<td [class.grid-container]="containerType === 'grid'" [colSpan]="detailColspan - 1">
+  <div class="details-title" *ngIf="rowItem?.detailTitleEnabled">{{rowItem?.detailsTitle}}</div>
+  <div class="resultdetaileditem">
+    <span *ngFor="let group of getGroups()">
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html:6]: Could the method be transformed into a pipe ?

File: projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
+      <span class="resultdetaileditem--group-title {{group.group.toLowerCase().replace(' ','-')}}-item-detail">{{group.group | translate}}</span>
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html:7]: Could the method be replaced by a pipe ?

File: projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html [hunk 4]
```diff
@@ -1,31 +1,35 @@
...
-      <a class="result-attachment-link" href="{{getAttachmentUrl(att.url)}}" matTooltip="{{getAttachmentDescription(att)}}"
-        [matTooltipPosition]="'after'" target="_blank">
-        <mat-icon *ngIf="att.icon" class="result-attachment-icon">{{att.icon}}</mat-icon>
-        <span class="result-attachment-label"> {{getAttachmentLabel(att, i + 1)}}</span>
-      </a>
+    <span class="result-attachments" *ngIf="rowItem?.attachments && rowItem?.attachments.length > 0">
+      <span class="result-attachments--title">{{'Attachments' | translate}}</span>
+      <br>
+      <span class="result-attachments--item" *ngFor="let att of rowItem?.attachments; index as i">
+        <a class="result-attachments--item--link" href="{{getAttachmentUrl(att.url)}}" matTooltip="{{getAttachmentDescription(att)}}"
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/results/result-detailed-item/result-detailed-item.component.html:26]: Could the methods be replaced by pipes ?

**Review by QuCMGisaia — APPROVED** [review]

File: projects/arlas-components/src/lib/components/results/model/item.ts (hunk 1 above)
Comment (sebbousquet) [reply projects/arlas-components/src/lib/components/results/model/item.ts]: Fixed

**Review by MAudelGisaia — APPROVED** [review]

File: projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts:1]: Missing licence

File: projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts:1]: missing licence

File: projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Injectable, Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'replace'
+})
+@Injectable()
+export class ReplacePipe implements PipeTransform {
+
+  public transform(value: string, searchValue: string, replaceValue: string): any {
+    return value.replace(searchValue, replaceValue);
```
[the remark this answers is not in the document]
Comment (sebbousquet) [reply projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts]: ✅

File: projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Injectable, Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'replace'
+})
+@Injectable()
+export class ReplacePipe implements PipeTransform {
+
+  public transform(value: string, searchValue: string, replaceValue: string): any {
```
[the remark this answers is not in the document]
Comment (sebbousquet) [reply projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts]: Done

File: projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts
Comment (sebbousquet) [reply projects/arlas-components/src/lib/pipes/replace/replace.pipe.ts:1]: added

File: projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts
Comment (sebbousquet) [reply projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts:1]: added

File: projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'getAttachmentUrl'
+})
+export class GetAttachmentUrlPipe implements PipeTransform {
+
+  public transform(value: string): unknown {
```
[the remark this answers is not in the document]
Comment (sebbousquet) [reply projects/arlas-components/src/lib/pipes/get-attachment-url/get-attachment-url.pipe.ts]: Done

**Review by QuCMGisaia — APPROVED** [review]

**Pull request merged**